kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
Much of that was written before we had a good sense of how to properly use vuex and deal with asynchronous behaviors. See e.g. https://trello.com/c/VOJvBYTn/718-user-edit-modal-error-handling